### DISCUSSION TITLE
Slow ohmyzsh start after PC restart

### DISCUSSION THREAD

**Discussion by erychagov** [ohmyzsh/ohmyzsh#10690] [Help]
Hello.
I notice that I have slow start of ohmyzsh.
I found that the next line takes about 20 secs on my PC:
```
zcompdump_revision="#omz revision: $(builtin cd -q "$ZSH"; git rev-parse HEAD 2>/dev/null)"
```
I found this line in the `$ZSH/oh-my-zsh.sh` file.
Why we need it? I want to find a  way to speedup it.

**Comment by rwmitchell** [comment]
Is it in a large git directory?

**Reply by erychagov** [reply]
I don't know. It's a default oh-my-zsh folder.

**Reply by rwmitchell** [reply]
I was referring to the directory you shell starts in.  Normally this is either your $HOME or the last directory you were working in.  Is zsh starting in $ZSH ?

$ZSH shouldn't be causing a 20 second lag in startup, it isn't a big git repo.
If a git repo has been created in $HOME, that might do it.

**Reply by erychagov** [reply]
ZSH starts at my $HOME folder. The folder takes a lot of space just because of OS.

**Reply by rwmitchell** [reply]
Hadn't had coffee yet and had only focused on the git command, not the preceding cd to $ZSH.

**Comment by carlosala** [comment]
> Hello.
> I notice that I have slow start of ohmyzsh.
> I found that the next line takes about 20 secs on my PC:
> ```
> zcompdump_revision="#omz revision: $(builtin cd -q "$ZSH"; git rev-parse HEAD 2>/dev/null)"
> ```
> I found this line in the `$ZSH/oh-my-zsh.sh` file.
> Why we need it? I want to find a  way to speedup it.

this line should not take more than 20ms in an old HDD. Could you please give git and Zsh version?

**Reply by erychagov** [reply]
The last commit is `02b52a03a5a78362c57d75c507240f69d4260d9a`
```
$> zsh --version
zsh 5.8 (x86_64-apple-darwin21.0)
```

I have SSD 1 Tb

**Reply by carlosala** [reply]
Which Git version do you have? (`git --version`)

**Reply by erychagov** [reply]
git version 2.32.0 (Apple Git-132)

**Reply by erychagov** [reply]
@carlosala Any ideas? :)

**Reply by mcornella** [reply]
Posted as an issue for improved readability https://github.com/ohmyzsh/ohmyzsh/issues/10693

**Comment by rwmitchell** [comment]
if you cd to $ZSH and then run
```time git rev-parse HEAD```
how long does it take?

**Reply by erychagov** [reply]
`git rev-parse HEAD  0.01s user 0.00s system 76% cpu 0.014 total`

**Reply by rwmitchell** [reply]
that looks normal.  how did you determine the offending line?  I'm assuming by adding echo or printf statements until you zeroed in on it?

**Reply by erychagov** [reply]
You're right! I used `echo` to find the line.

**Reply by rwmitchell** [reply]
Does this happen on each new terminal window?

Do you see the same slow behavior when running
```exec zsh```

**Reply by erychagov** [reply]
> Does this happen on each new terminal window?

Only first terminal window takes time

> Do you see the same slow behavior when running

No.

**Reply by mcornella** [reply]
Posted as an issue for improved readability https://github.com/ohmyzsh/ohmyzsh/issues/10693
